README.md

```diff
@@ -1,44 +1,57 @@
-- `sam.test.yml` in the root of v1 and v2. For v1, this should be identical to `sam.qa.yml` in discovery-hybrid-indexer, with one change: replace the value for ELASTIC_SEARCH_RESOURCES_INDEX with test-index-{your initials}-v1 or -v2
+### 1. Configure the Tester
+
+Create your own `.env`:
+
+```
+cp .env-sample .env
+```
+
+And edit `TEST_INDEX` to use your initials.
+
+### 2. Prepare the indexers
 
-### Setting up environment:
-0. `nvm use && npm i`
 1. Clone the latest v1 discovery-hybrid-indexer into the root of this directory. Rename the directory v1
 2. Clone the new v2 indexer you are testing into a different directory. Name this directory v2
 3. Create v1/sam.test.yml . Copy the contents of sam.qa.yml into it, and replace ELASTIC_RESOURCES_INDEX with test-index-{your initials}-v1. Make sure there is a v2/sam.test.yml that has the necessary config, along with ELASTIC_RESOURCES_INDEX = test-index-{your initials}-v2.
 4. Create a blank Elastic Search indexes for v1 and v2 to write to: `npm run recreate-es-indexes`
-5. Install v1 and v2 as necessary
+5. Install v1 and v2 as necessary (i.e. `cd v1; nvm use; npm i; cd ../v2; nvm use; npm i`)
+
+### 3. Prepare test data
+
+To update `uris.csv`, export from this google sheet https://docs.google.com/spreadsheets/d/1a2QKIsRJrEPel2K5znzxnwoREzDWJOMZ6nIfz2EsDRg/edit#gid=0 and save in the root as `uris.csv`
 
-### URIs for indexing
-To update uris.csv, export from this google sheet https://docs.google.com/spreadsheets/d/1a2QKIsRJrEPel2K5znzxnwoREzDWJOMZ6nIfz2EsDRg/edit#gid=0 and save in the root as uris.csv
+Rebuild event files:
+
+`npm run rebuild-events`
+
+If faced with the error: `Unable to load module 'csv-parse/sync'`, try running `nvm use`.
+
+### 4. Populate the indexes
 
-### Update events 
-To rebuild unencoded records and encoded events, run `npm run rebuild-events`. If faced with the error: `Unable to load module 'csv-parse/sync'`, try running `nvm use`.
-### Running tests
 1. Start up local lambda for v1: `npm run start-lambda-v1`
 2. In a separate terminal, send the events to the lambda: `npm run send-events-v1`. 
 3. Repeat steps 1+2 replacing v1 with v2 in the scripts
-4. Check that the ES indices are identical: `npm test`
 
-### Cleanup
-To delete Elastic Indexes created, run `node cleanup-elastic-index` 
+### 5. Test parity
 
-### Troubleshooting
-- make sure discovery-api-indexer is at most recent version in root level package.json
-- 
+Check that the ES indices are identical: `npm test`
 
+### Cleanup
 
+To delete Elastic Indexes created, run `node cleanup-elastic-index`
 
+### Troubleshooting
 
+Getting "strict mapping" errors when writing documents to ES?
+- Make sure discovery-api-indexer is at most recent version in root level package.json. You may need to run `npm i github:NYPL-discovery/discovery-api-indexer#v1.8.0` (or whatever the latest tag is) manually in the Terminal to force an install of the latest tag.
```

Comment:
what is this error?

Reply:
If an indexer candidate attempts to write properties to the index that haven't been explicitly added to the ES mapping, ES will throw an error. This is because we've intentionally overridden the ["dynamic" mapping config](https://www.elastic.co/guide/en/elasticsearch/reference/5.3/dynamic.html). We do this because we want to be intentional about the data we write and avoid the ES feature of intuiting datatypes based on the first data sample it sees, which works well enough until it doesn't, and then you're stuck with a mapping that doesn't fit later data.

Reply:
ok. we should probably call out the index schema in the RCI for troubleshooting as well.

Reply:
I added a note with a link to the RCI index schema definition